src/main/java/frc/robot/subsystems/Lift.java

```diff
@@ -1,36 +1,159 @@
+ * 
+ * How Is It Triggered? 1) User Input
+ * 
+ * Does It Store Any State? 1) isBraked 2) Limit switch values 3) Encoder values
+ * 
+ * Sensors? 1) Encoders 2) Limit Switches
+ */
 
-    private SendableSpeedController elevatorMotor;
+public class Lift extends SubsystemBase {
 
+    private PIDSpeedController elevatorMotor;
+    private ISolenoid elevatorBrake;
+    private IEncoder liftEncoder;
     private static final double elevatorMotorSpeed = 1;
 
+    boolean isBraked = elevatorBrake.get();
+
+    // TO DO Find a way to sync the elevatorLeft and elevatorRight motors
     public Lift(RobotMap.LiftMap map) {
         super();
         elevatorMotor = map.elevator();
+        liftEncoder = map.getLiftEncoder();
+        elevatorBrake = map.liftBrake();
+    }
+
+    // TO DO Confgure encoders to actually read a position
+    int getPosition = liftHeights.Bottom.value();
+
+    // TODO make these heights reflect where we actually want them to be
+    public enum liftHeights {
+        Top(3), Middle(2), Bottom(1);
+
+        private int iPosition;
+
+        // Returning an interger to compare whether we're in the right place or not
+        private int value() {
+            return this.iPosition;
+        }
+
+        // Returning the level the lift is at (top middle or bottom)
+        private liftHeights(int iPosition) {
+            this.iPosition = iPosition;
+        }
+    }
+
+    public CommandBase moveLift(DoubleSupplier speed) {
+        // The command is named "Move Lift" and requires this subsystem.
+        return new CommandBase() {
+
+            @Override
+            public void initialize() {
+                disengageBrake();
+
+            }
+
+            @Override
+            public boolean isFinished() {
+                return false;
+            }
+
+            @Override
+            public void execute() {
+                elevatorMotor.set(speed.getAsDouble());
+
+            }
+
+            @Override
+            public void end(final boolean interrupted) {
+                engageBrake();
+
+            }
+
+        };
+    }
+
+    public InstantCommand engageBrake() {
+        return new InstantCommand(() -> {
+            elevatorBrake.set(true);
+        });
     }
 
-    public CommandBase up() {
-        return new StartEndCommand(() -> {
-            elevatorMotor.set(-elevatorMotorSpeed);
-        }, () -> {
-            elevatorMotor.stopMotor();
-        }, this);
+    public InstantCommand disengageBrake() {
+        return new InstantCommand(() -> {
+            elevatorBrake.set(false);
+        });
     }
 
-    public CommandBase down() {
-        return new StartEndCommand(() -> {
-            elevatorMotor.set(elevatorMotorSpeed);
-        }, () -> {
-            elevatorMotor.stopMotor();
-        }, this);
+    public InstantCommand toggleBrake() {
+        return new InstantCommand(() -> {
+            if (isBraked == false) {
```

Comment:
- This command doesn't use the subsystem.
- Use `elevatorBrake.set(!isBraked)`
- Make sure you update `elevatorBrake` here and elsewhere to keep the value accurate.